eureka2-integration/src/test/java/com/netflix/eureka2/integration/client/EurekaClientFailoverTest.java

```diff
@@ -87,29 +87,35 @@ public void run() {
     }
 
     private void executeFailoverTest(Runnable failureInjector) throws Exception {
-        // Subscribe to instance registration updates
-        ExtTestSubscriber<ChangeNotification<InstanceInfo>> interestSubscriber = subscribeTo(INSTANCE_UP);
-
-        // Register
-        PublishSubject<InstanceInfo> registrationSubject = PublishSubject.create();
         EurekaRegistrationClient registrationClient = eurekaDeploymentResource.getEurekaDeployment().registrationClientToWriteCluster();
-        Subscription registrationSubscription = registrationClient.register(registrationSubject).subscribe();
-        registrationSubject.onNext(INSTANCE_UP);
+        EurekaInterestClient interestClient = eurekaDeploymentResource.getEurekaDeployment().cannonicalInterestClient();
+
+        try {
+            // Subscribe to instance registration updates
+            ExtTestSubscriber<ChangeNotification<InstanceInfo>> interestSubscriber = subscribeTo(interestClient, INSTANCE_UP);
 
-        assertThat(interestSubscriber.takeNext(60, TimeUnit.SECONDS), is(addChangeNotificationOf(INSTANCE_UP)));
+            // Register
+            PublishSubject<InstanceInfo> registrationSubject = PublishSubject.create();
+            Subscription registrationSubscription = registrationClient.register(registrationSubject).subscribe();
+            registrationSubject.onNext(INSTANCE_UP);
 
-        // Inject failure
-        failureInjector.run();
+            assertThat(interestSubscriber.takeNext(60, TimeUnit.SECONDS), is(addChangeNotificationOf(INSTANCE_UP)));
 
-        // Update instance status, and verify that it was handled
-        registrationSubject.onNext(INSTANCE_DOWN);
+            // Inject failure
+            failureInjector.run();
 
-        assertThat(registrationSubscription.isUnsubscribed(), is(false));
-        assertThat(interestSubscriber.takeNext(60, TimeUnit.SECONDS), is(modifyChangeNotificationOf(INSTANCE_DOWN)));
+            // Update instance status, and verify that it was handled
+            registrationSubject.onNext(INSTANCE_DOWN);
+
+            assertThat(registrationSubscription.isUnsubscribed(), is(false));
+            assertThat(interestSubscriber.takeNext(60, TimeUnit.SECONDS), is(modifyChangeNotificationOf(INSTANCE_DOWN)));
+        } finally {
+            registrationClient.shutdown();
```

Comment:
There seemed to be some issues with these test when run together (one of them fails, but both passes fine when run independently). This points to resource clean up issues and this try finally fixes this.